src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseRegexMembers.cs

```diff
@@ -0,0 +1,155 @@
+            CreateLocalizableResourceString(nameof(UseRegexCountMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(UseRegexCountDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
+            UseRegexIsMatchRuleId,
+            UseRegexCountRuleId);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(context =>
+            {
+                // Require that Regex and supporting types exist.
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextRegularExpressionsGroup, out var groupType) ||
+                    !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextRegularExpressionsMatchCollection, out var matchCollectionType) ||
+                    !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextRegularExpressionsRegex, out var regexType))
+                {
+                    return;
+                }
+
+                // Get the various members needed from Regex types.
+                var groupSuccessSymbol = groupType.GetMembers("Success").FirstOrDefault();
+                var matchCollectionCountSymbol = matchCollectionType.GetMembers("Count").FirstOrDefault();
+                var regexMatchSymbols = regexType.GetMembers("Match");
+                var regexMatchesSymbols = regexType.GetMembers("Matches");
+                var regexIsMatchSymbols = regexType.GetMembers("IsMatch");
+                var regexCountSymbols = regexType.GetMembers("Count");
+
+                // Ensure we have the required member symbols (Regex.Count is optional).
+                if (groupSuccessSymbol is null ||
+                    matchCollectionCountSymbol is null ||
+                    regexMatchSymbols.Length == 0 ||
+                    regexMatchesSymbols.Length == 0 ||
+                    regexIsMatchSymbols.Length == 0)
+                {
+                    return;
+                }
+
+                // Everything we're looking for is a property, so find all property references.
+                context.RegisterOperationAction(context =>
+                {
+                    var initialPropRef = (IPropertyReferenceOperation)context.Operation;
+
+                    // Regex.Match(...).Success. Look for Group.Success property access.
+                    if (SymbolEqualityComparer.Default.Equals(initialPropRef.Property, groupSuccessSymbol))
+                    {
+                        if (initialPropRef.Instance is IInvocationOperation regexMatchInvocation &&
```

Comment:
Does this work both for `Regex.Match().Success` as well as `myRegex.Match().Success`?

Reply:
Looks like yes, just saw your tests :)